src/snowflake/cli/plugins/nativeapp/artifacts.py

```diff
@@ -283,3 +299,37 @@ def find_version_info_in_manifest_file(
             patch_name = version_info[patch_field]
 
     return version_name, patch_name
+
+
+def project_path_to_deploy_path(
+    project_paths: List[Path], files_mapping: DeployMapping
+) -> List[Path]:
+    """Given a list of source paths, returns the deploy destination paths. This function assumes that a build was performed before calling it."""
+
+    def calculate_deploy_path(project_path: str) -> Path:
+        # Find a common directory that exists under the deploy directory
+        root = Path(project_path)
+        while root:
+            if str(root) in files_mapping:
+                break
+            elif root.parent != root:
+                root = root.parent
+            else:
+                raise FileNotFoundError(project_path)
+
+        # Construct the target deploy path
+        path_to_symlink = files_mapping[str(root)]
+        path_to_target = Path(project_path).relative_to(root)
+        result = Path(path_to_symlink, path_to_target)
+
+        if not result.exists():
+            raise FileNotFoundError(result)
+        return result
+
+    deploy_paths: List[Path] = []
+    for project_path in map(str, project_paths):
+        if project_path in files_mapping:
```

Comment:
generally I find it confusing and error-prone that we use strings for project paths. Depending on how these paths are obtained, they may not be in canonical form, but the logic seems to assume they are. Can we make our logic more resilient?